### PR TITLE
scrub: support scrub_mode

### PR DESCRIPTION
**Pull request by bhalevy**
Add support for the scrub_mode option for the scylla api.

Signed-off-by: Benny Halevy <bhalevy@scylladb.com>

### CODE REVIEW

File: src/main/java/org/apache/cassandra/service/StorageServiceMBean.java [hunk 1]
```diff
@@ -364,10 +364,15 @@ public int scrub(boolean disableSnapshot, boolean skipCorrupted, boolean checkDa
...
             String... columnFamilies) throws IOException, ExecutionException, InterruptedException;
 
+    @Deprecated
     public int scrub(boolean disableSnapshot, boolean skipCorrupted, boolean checkData, boolean reinsertOverflowedTTL,
             int jobs, String keyspaceName, String... columnFamilies)
             throws IOException, ExecutionException, InterruptedException;
 
+    public int scrub(boolean disableSnapshot, boolean skipCorrupted, String scrubMode, boolean checkData, boolean reinsertOverflowedTTL,
+            int jobs, String keyspaceName, String... columnFamilies)
+            throws IOException, ExecutionException, InterruptedException;
```
Comment (denesb) [inline src/main/java/org/apache/cassandra/service/StorageServiceMBean.java:374]: I think there should be a variant with `skipCorrupted` and one with `scrubMode`, doesn't really make sense to set both, as they conflict (and I don't recall how the conflict is resolved in the API).

Comment (bhalevy) [reply src/main/java/org/apache/cassandra/service/StorageServiceMBean.java:374]: Hmm, that makes sense.  Our documentation says that scrubMode trumps skipCorrupted.
https://github.com/scylladb/scylla/blob/f98cb965067b8a8f640840387dcbf043a0885d8c/api/api-doc/storage_service.json#L798

Comment (bhalevy) [reply src/main/java/org/apache/cassandra/service/StorageServiceMBean.java:378]: Fixed in v2

**Comment by bhalevy** [comment]
In v2:
- deprecate skipCorrupted, replace the skipCorrupted parameter with scrubMode in new api method.

**Review by denesb — APPROVED** [review]

**Comment by bhalevy** [comment]
@scylladb/scylla-jmx-maint please merge

**Pull request closed without merging**